### PR TITLE
add an option -H to samtools depth to print a file header

### PR DESCRIPTION
**Pull request by lindenb**
This PR adds a new option `-H` to samtools depth to print a header.

with filenames

```
$ ./samtools depth -H ~/src/jvarkit-git/src/test/resources/S*.bam | head -n 3 
#CHROM	POS	/home/lindenb/src/jvarkit-git/src/test/resources/S1.bam	/home/lindenb/src/jvarkit-git/src/test/resources/S2.bam	/home/lindenb/src/jvarkit-git/src/test/resources/S3.bam	/home/lindenb/src/jvarkit-git/src/test/resources/S4.bam	/home/lindenb/src/jvarkit-git/src/test/resources/S5.bam
RF01	1	1	0	0	0	0
RF01	2	1	0	0	0	0
```
with stdin

```
$ cat ~/src/jvarkit-git/src/test/resources/S1.bam | ./samtools depth -H - | head
#CHROM	POS	-
RF01	1	1
RF01	2	1
RF01	3	1
RF01	4	1
RF01	5	1
RF01	6	1
RF01	7	1
RF01	8	2
RF01	9	2
```

### CODE REVIEW

**Comment by lindenb** [comment]
UPDATE: I fixed the formatting (was https://twitter.com/jomarnz/status/1036380562594955269 ) and added an option -o to specify the output filename.

check the error message

```
$ cat ~/src/jvarkit-git/src/test/resources/S1.bam | ./samtools depth -o /usr/bin/cannotwrite -
samtools depth: Cannot open "/usr/bin/cannotwrite" for writing
: Permission denied
```

save to stdout

```
$ cat ~/src/jvarkit-git/src/test/resources/S1.bam | ./samtools depth -o - - | head
RF01	1	1
RF01	2	1
RF01	3	1
RF01	4	1
RF01	5	1
RF01	6	1
RF01	7	1
RF01	8	2
RF01	9	2
RF01	10	2
```

save to a file

```
$ cat ~/src/jvarkit-git/src/test/resources/S1.bam | ./samtools depth -o jeter.txt -  && head jeter.txt
RF01	1	1
RF01	2	1
RF01	3	1
RF01	4	1
RF01	5	1
RF01	6	1
RF01	7	1
RF01	8	2
RF01	9	2
RF01	10	2
```

**Comment by jmarshall** [comment]
I don't think you've pushed the update :smile:

`print_error_errno()` writes the newline itself, so the error message string shouldn't end with `\n`.

Another suggestion is to print the entire header inside one `if (print_header) { … }` at the bottom, where it currently does the `\n` printing. That way if there's an error opening one of the input files, you won't have printed out half a header line.

**Comment by lindenb** [comment]
@jmarshall 

> I don't think you've pushed the update

Ah, you're right, it's one of those awful "HEAD detached from origin/pl_stdepth_header"... :sob:

**Comment by lindenb** [comment]
@jmarshall thanks + back to you, I've moved the print_header block after the  'open loop'.

File: bam2depth.c [hunk 1]
```diff
@@ -154,7 +168,7 @@ int main_depth(int argc, char *argv[])
         n = argc - optind; // the number of BAMs on the command line
     data = calloc(n, sizeof(aux_t*)); // data[i] for the i-th input
     reg_tid = 0; beg = 0; end = INT_MAX;  // set the default region
-    if(print_header) fputs("#CHROM\tPOS",stdout);
+    if( print_header ) fputs("#CHROM\tPOS", file_out);
```
Comment (jmarshall) [inline bam2depth.c]: Look at the existing code nearby in this function. What I was asking for is `if (print_header)`, not this `if( foo )` abomination.

Comment (jkbonfield) [reply bam2depth.c]: Sadly the `if ( foo )` notation is commonly used too in our code. Bah.

Some counts, as I was curious:

`if(x)   ` 16
`if( x )  ` 3
`if (x)  ` 4873
`if ( x )` 1165

The One True Way is fortunately winning the battle. :-)

Comment (jmarshall) [reply bam2depth.c]: Yes. It's Petr's preferred style, and I was never able to get him to compromise and use the local style when he was working on One True Way-formatted files — so the infection has made inroads outwith bcftools.

**Comment by lindenb** [comment]
@jmarshall fixed + thanks

**Comment by daviesrob** [comment]
Thanks.  Squashed, rebased and merged as a581b9d.  I've also updated the manual page and put in a check for `fclose()` failure (in 0bf3778).

**Pull request closed without merging**